### DISCUSSION TITLE
Adding the .NET Aspire orchestration code failed. Failed to insert code for project X. Can't find a call to method(s) Build on type Microsoft.AspNetCore.Builder.WebApplicationBuilder in the program main

### DISCUSSION THREAD

**Discussion by krzysiek-b** [microsoft/aspire#4474] [Q&A]
I have 
```
		WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
		WebApplication _ = builder.Build();
```
in program.cs. What do you require here?

**Comment by davidfowl** [comment]
@antonfirsov @vijayrkn Seems like the tooling failed for this pattern?

**Reply by antonfirsov** [reply]
@davidfowl I assume you wanted to tag someone else?

**Reply by davidfowl** [reply]
Oops 😅

**Comment by krzysiek-b** [comment]
What I see if I do so
```
int Main()
{
    WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
    WebApplication _ = builder.Build();
}
```
it works but it breaks on
```
// file: program.cs
int Main()
{
    Configure(); 
}

void Configure()
{
   WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
   WebApplication _ = builder.Build();
}
```

I assume this was my issue, but do you have plans to support this?

**Reply by BillHiebert** [reply]
As you noted it is currently not supported. We assume the call to CreateBuilder() is in the startup method.  I've added an item to our backlog to look at improving this.

**Comment by rsalus** [comment]
So, if I understand correctly, the `CreateBuilder()` and `Build()` calls need to be within `Main()`. This is typically the case when using the minimal program.cs pattern. However, if we’re converting a legacy app that follows an older pattern, those patterns aren’t supported at this time. Is that correct?

EX (non-functional with Aspire):
![image](https://github.com/user-attachments/assets/31bec9d4-2417-46cf-80ed-7b5e6399e04f)

**Comment by BillHiebert** [comment]
Yes, it has to be the minimal program.cs pattern. However, the configuration can be done in a method that is called from program.cs and we should find it.
